File: LR2/11.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = 0, M = 0;
        String S = "";

        try {
            // Чтение N и M из ввода
            N = scanner.nextInt();
            M = scanner.nextInt();
            // Чтение строки S (команды)
            S = scanner.next();
        } catch (Exception e) {
            // Если ввод не корректен (не числа или нет данных)
            System.out.println("Invalid input");
            scanner.close();
            return;
        }

        // Проверка, что размеры доски корректны (N и M >=1)
        if (N < 1 || M < 1) {
            System.out.println("Invalid input");
            scanner.close();
            return;
        }

        // Проверка символов в строке S на допустимость
        for (char c : S.toCharArray()) {
            if (c != 'L' && c != 'R' && c != 'U' && c != 'D') {
                System.out.println("Invalid input");
                scanner.close();
                return;
            }
        }

        int min_x = 1, min_y = 1;
        int L = 0, R = 0, U = 0, D = 0;

        // Обработка каждой команды с динамическим обновлением координат
        for (char c : S.toCharArray()) {
            switch (c) {
                case 'L':
                    L++;
                    if (min_x - L < 1) min_x++;
                    if (R > 0) R--;
                    break;
                case 'R':
                    R++;
                    if (L > 0) L--;
                    break;
                case 'U':
                    U++;
                    if (min_y - U < 1) min_y++;
                    if (D > 0) D--;
                    break;
                case 'D':
                    D++;
                    if (U > 0) U--;
                    break;
            }
        }

        // Проверка существования допустимой начальной позиции
        if (min_x > M || min_y > N) {
            System.out.println("No");
        } else {
            // Вывод минимальных координат
            System.out.println("(" + min_x + "," + min_y + ")");
        }

        scanner.close();
    }
}

File: LR2/11.php
<?php
// Чтение входных данных из стандартного ввода
$input = trim(fgets(STDIN));
$parts = explode(' ', $input);

// Проверка наличия трёх элементов (N, M, S)
if (count($parts) < 3) {
    echo "Invalid input\n";
    exit;
}

list($nStr, $mStr, $s) = $parts;

// Проверка, что N и M являются целыми числами
if (!ctype_digit($nStr) || !ctype_digit($mStr)) {
    echo "Invalid input\n";
    exit;
}

$n = intval($nStr);
$m = intval($mStr);

// Проверка допустимости размеров доски
if ($n < 1 || $m < 1) {
    echo "Invalid input\n";
    exit;
}

// Проверка допустимости символов в строке команд
foreach (str_split($s) as $c) {
    if (!in_array($c, ['L', 'R', 'U', 'D'])) {
        echo "Invalid input\n";
        exit;
    }
}

// Подсчёт количества каждой команды
$l = $r = $u = $d = 0;
$min_x = 1;
$min_y = 1;

foreach (str_split($s) as $c) {
    switch ($c) {
        case 'L': 
            $l++;
            if ($min_x - $l < 1) $min_x++;
            if ($r > 0) $r--;
            break;
        case 'R': 
            $r++;
            if ($l > 0) $l--;
            break;
        case 'U': 
            $u++;
            if ($min_y - $u < 1) $min_y++;
            if ($d > 0) $d--;
            break;
        case 'D': 
            $d++;
            if ($u > 0) $u--;
            break;
    }
}

// Проверка существования допустимой начальной позиции
if ($min_x > $m || $min_y > $n) {
    echo "No\n";
} else {
    echo "($min_x,$min_y)\n";
}
?>